### ISSUE TITLE
Blank characters in RPres above slide separator causing empty heading tags

### ISSUE THREAD

**Issue by cgrudz** [rstudio/rstudio#5444]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.5001
    OS Version      :  Ubuntu 18.04.3 LTS
    R Version       :  3.4.4 (2018-03-15)

### Steps to reproduce the problem
Using .rpres markdown templates, if there are blank characters above the slide separator 

```
========================================================
```
this will introduce an empty heading tag in the HTML output.   See, e.g., the .rpres template below:
```{markdown}
H3 bug
========================================================
date: 09/20/2019
autosize: true

  
========================================================
### The h3 tag I want

* foo
```
when output to html, the html file will have an empty ```<h3> </h3>``` above "The h3 tag I want".

### Describe the problem in detail
This may not be a bug strictly speaking, but it can introduce various issues.  Particularly, if a user is using descending heading tags for sections of a presentation as per web accessibility guidelines, these should be introduced manually in the slide without skipped heading levels.  However, if there are blank spaces above the slide separator, this will introduce blank headings which interfere with screen readers.

### Describe the behavior you expected
It would be preferable for the compiler to remove blank headings automatically, so as not to interfere with screen readers, if these are introduced by accident.
